### PR TITLE
classical vs fine-grained github tokens

### PR DESCRIPTION
**Pull request by patkearns10**
## What are you changing in this pull request and why?

Having some confusion with customers re: this recent (breaking) fix: https://docs.getdbt.com/docs/dbt-versions/release-notes/dec-2022/private-packages-clone-git-token

Have had some customers claim that they require user and some that do not require user. I think it is because of classical vs fine-grained personal access tokens: https://stackoverflow.com/questions/74532852/github-clone-repo-with-fine-grained-personal-access-tokens-pat

Screen Shot 2023-02-10 at 3 25 28 PM

This [customer](https://dbtcloud.zendesk.com/agent/tickets/40834) did not need to use the username field 
This [customer](https://dbtcloud.zendesk.com/agent/tickets/41890) did need to use the username field

[slack discussion](https://dbt-labs.slack.com/archives/CH1AWA9ST/p1675965418338379?thread_ts=1672781028.683789&cid=CH1AWA9ST)

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

**Comment by mirnawong1** [comment]
hey @patkearns10 thanks so much for opening this up - great addition! i recently removed the username in this pr: #2797 but it seems username is required for fine grained permissions then right?

File: website/docs/docs/build/packages.md [hunk 1]
```diff
@@ -205,8 +205,11 @@ packages:
   # use this format when accessing your repository via a github application token
   - git: "https://{{env_var('DBT_ENV_SECRET_GIT_CREDENTIAL')}}@github.com/dbt-labs/awesome_repo.git" # git HTTPS URL
 
-  # use this format when accessing your repository via a personal access token
+  # use this format when accessing your repository via a classical personal access token
   - git: "https://{{env_var('DBT_ENV_SECRET_GIT_CREDENTIAL')}}@github.com/dbt-labs/awesome_repo.git" # git HTTPS URL
+ 
+   # use this format when accessing your repository via a fine-grained personal access token (username sometimes required)
+  - git: "https://<username>:{{env_var('DBT_ENV_SECRET_GIT_CREDENTIAL')}}@github.com/dbt-labs/awesome_repo.git" # git HTTPS URL
```
Comment (mirnawong1) [inline website/docs/docs/build/packages.md]: @patkearns10 confirming you mean github_username here - just making it extra clear for users:

```suggestion
  - git: "https://GITHUB_USERNAME:{{env_var('DBT_ENV_SECRET_GIT_CREDENTIAL')}}@github.com/dbt-labs/awesome_repo.git" # git HTTPS URL
```

**Review by mirnawong1 — APPROVED** [review]
thanks for adding this to clarify our Docs @patkearns10 - made a small tweak and approved for you to merge! 

these git tokens have really been tripping people up!

**Review by schottj — COMMENTED** [review]
lgtm! thanks for the quick fix

**Comment by patkearns10** [comment]
@mirnawong1 - I thought username was not required at the time you made that initial PR. But we have since learned it is because of fine-grained personal access tokens. 

All good on my end with the tweaks. TY!

**Pull request merged**